cmd/argocd/commands/app.go

```diff
@@ -1451,6 +1479,24 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 						},
 					}
 				}
+				if diffChanges {
+					app, err := appIf.Get(context.Background(), &applicationpkg.ApplicationQuery{Name: &appName})
+					errors.CheckError(err)
+					resources, err := appIf.ManagedResources(context.Background(), &applicationpkg.ResourcesQuery{ApplicationName: &appName})
+					errors.CheckError(err)
+					conn, settingsIf := acdClient.NewSettingsClientOrDie()
+					defer argoio.Close(conn)
+					argoSettings, err := settingsIf.Get(context.Background(), &settingspkg.SettingsQuery{})
+					errors.CheckError(err)
+					foundDiffs := false
+					findandPrintDiff(app, resources, argoSettings, appName, &foundDiffs, diffOption)
+					if foundDiffs && !diffChangesConfirm {
+						yesno := cli.AskToProceed("Confirm to Sync Application y/n :")
```

Comment:
```suggestion
						yesno := cli.AskToProceed("Please review changes to Application resources shown above. Do you want to continue the sync process (y/n) :")
```

Reply:
@jannfis : if multiple apss are passed? should we ask for confirmation for each app sync? I think since its a single command request no need to ask for each app seperately

Reply:
I think it already asks for confirmation of each app sync, at least during my tests, this was the behavior. One thing I noted tho, it just shows the differences and asks for confirmation without any context, which may be confusing. For example:

```
$ ./dist/argocd app sync guestbook guestbook3 --preview-changes
===== /Service guestbook/kustomize-guestbook-ui ======
0a1,13
> apiVersion: v1
> kind: Service
> metadata:
>   labels:
>     app.kubernetes.io/instance: guestbook
>   name: kustomize-guestbook-ui
>   namespace: guestbook
> spec:
>   ports:
>   - port: 80
>     targetPort: 80
>   selector:
>     app: guestbook-ui
Please review changes to Application resources shown above. Do you want to continue the sync process? (y/n) :y
TIMESTAMP                  GROUP        KIND   NAMESPACE                  NAME      STATUS    HEALTH        HOOK  MESSAGE
2022-03-01T10:23:07+00:00            Service   guestbook  kustomize-guestbook-ui  OutOfSync  Missing               
2022-03-01T10:23:07+00:00   apps  Deployment   guestbook  kustomize-guestbook-ui    Synced   Healthy               
2022-03-01T10:23:07+00:00            Service   guestbook  kustomize-guestbook-ui  OutOfSync  Missing              service/kustomize-guestbook-ui created
2022-03-01T10:23:07+00:00   apps  Deployment   guestbook  kustomize-guestbook-ui    Synced   Healthy              deployment.apps/kustomize-guestbook-ui unchanged
2022-03-01T10:23:07+00:00            Service   guestbook  kustomize-guestbook-ui    Synced  Healthy              service/kustomize-guestbook-ui created

Name:               guestbook
Project:            default
Server:             https://kubernetes.default.svc
Namespace:          guestbook
URL:                http://127.0.0.1:8080/applications/guestbook
Repo:               https://github.com/argoproj/argocd-example-apps
Target:             HEAD
Path:               kustomize-guestbook
SyncWindow:         Sync Allowed
Sync Policy:        <none>
Sync Status:        Synced to HEAD (53e28ff)
Health Status:      Healthy

Operation:          Sync
Sync Revision:      53e28ff20cc530b9ada2173fbbd64d48338583ba
Phase:              Succeeded
Start:              2022-03-01 10:23:07 +0000 UTC
Finished:           2022-03-01 10:23:07 +0000 UTC
Duration:           0s
Message:            successfully synced (all tasks run)

GROUP  KIND        NAMESPACE  NAME                    STATUS  HEALTH   HOOK  MESSAGE
       Service     guestbook  kustomize-guestbook-ui  Synced  Healthy        service/kustomize-guestbook-ui created
apps   Deployment  guestbook  kustomize-guestbook-ui  Synced  Healthy        deployment.apps/kustomize-guestbook-ui unchanged
===== /Service guestbook3/kustomize-guestbook-ui ======
0a1,13
> apiVersion: v1
> kind: Service
> metadata:
>   labels:
>     app.kubernetes.io/instance: guestbook3
>   name: kustomize-guestbook-ui
>   namespace: guestbook3
> spec:
>   ports:
>   - port: 80
>     targetPort: 80
>   selector:
>     app: guestbook-ui
Please review changes to Application resources shown above. Do you want to continue the sync process? (y/n) :y
TIMESTAMP                  GROUP        KIND   NAMESPACE                  NAME      STATUS    HEALTH        HOOK  MESSAGE
2022-03-01T10:23:12+00:00            Service  guestbook3  kustomize-guestbook-ui  OutOfSync  Missing               
2022-03-01T10:23:12+00:00   apps  Deployment  guestbook3  kustomize-guestbook-ui    Synced   Healthy               
2022-03-01T10:23:12+00:00            Service  guestbook3  kustomize-guestbook-ui  OutOfSync  Missing              service/kustomize-guestbook-ui created
2022-03-01T10:23:12+00:00   apps  Deployment  guestbook3  kustomize-guestbook-ui    Synced   Healthy              deployment.apps/kustomize-guestbook-ui unchanged
2022-03-01T10:23:12+00:00            Service  guestbook3  kustomize-guestbook-ui    Synced  Healthy              service/kustomize-guestbook-ui created

Name:               guestbook3
Project:            default
Server:             https://kubernetes.default.svc
Namespace:          guestbook3
URL:                http://127.0.0.1:8080/applications/guestbook3
Repo:               https://github.com/argoproj/argocd-example-apps
Target:             HEAD
Path:               kustomize-guestbook
SyncWindow:         Sync Allowed
Sync Policy:        <none>
Sync Status:        Synced to HEAD (53e28ff)
Health Status:      Healthy

Operation:          Sync
Sync Revision:      53e28ff20cc530b9ada2173fbbd64d48338583ba
Phase:              Succeeded
Start:              2022-03-01 10:23:12 +0000 UTC
Finished:           2022-03-01 10:23:12 +0000 UTC
Duration:           0s
Message:            successfully synced (all tasks run)

GROUP  KIND        NAMESPACE   NAME                    STATUS  HEALTH   HOOK  MESSAGE
       Service     guestbook3  kustomize-guestbook-ui  Synced  Healthy        service/kustomize-guestbook-ui created
apps   Deployment  guestbook3  kustomize-guestbook-ui  Synced  Healthy        deployment.apps/kustomize-guestbook-ui unchanged
```
IMHO, we should provide some more context for the user, before showing the actual diff. For example "Showing differences between live and desired state of application 'guestbook':", and maybe, also include the application name in the prompt?

Reply:
yes i agree, should we include context in app diff command too?

Reply:
I think the context with the `app diff` command is implicit, because it only accepts a single application as argument. OTOH, with the `app sync` command, you can specify more than one application to sync and will be presented with diffs for multiple applications eventually. So I'd tend to say it's not needed for `app diff`.